### PR TITLE
varnish5: fix build

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
fixing some regression from #66559

varnish5 isn't ready for the upgrade to python3, but varnish 6 requires it, so i parameterized it in the generic builder.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @fpletz  @mmahut 

before:
```
[1 built (1 failed), 21 copied (28.5 MiB), 19.5 MiB DL]
error: build of '/nix/store/zjv7yyzyjbj4f0zf9kzwc9ifj9m42mc0-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/66559
4 package failed to build:
collectd collectd-data prometheus-varnish-exporter varnish
```
after:
```
[1 built, 6 copied (15.8 MiB), 4.0 MiB DL]
5 package were build:
collectd collectd-data prometheus-varnish-exporter varnish varnish4
```

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build varnish4 varnish5 varnish6

**Comment by mmahut** [comment]
@GrahamcOfBorg build varnish4 varnish5 varnish6

File: pkgs/servers/varnish/default.nix [hunk 1]
```diff
@@ -13,8 +13,8 @@ let
 
       nativeBuildInputs = [ pkgconfig ];
       buildInputs = [
-        pcre libxslt groff ncurses readline python3 libedit
-        python3.pkgs.docutils makeWrapper
+        pcre libxslt groff ncurses readline python libedit
+        python.pkgs.docutils makeWrapper
       ] ++ extraBuildInputs;
```
Comment (volth) [inline pkgs/servers/varnish/default.nix]: `python`, `python.pkgs.docutils`, `makeWrapper` seem to be in `nativeBuildInputs` ?

Comment (mmahut) [reply pkgs/servers/varnish/default.nix]: @jonringer do you mind addressing this? I will merge it afterwards.

Comment (jonringer) [reply pkgs/servers/varnish/default.nix]: Yea, i was also looking into some of the plugins, they seem to be in an odd state

**Comment by jonringer** [comment]
passed the python version used by varnish to the plugins, should work now

**Comment by jonringer** [comment]
@GrahamcOfBorg build varnish5Packages varnish4Packages

**Comment by volth** [comment]
I'd say `python` itself is `nativeBuildInputs` too as it used to produce docs.
But let's check

**Comment by volth** [comment]
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6Packages

**Comment by jonringer** [comment]
it was for some of the packages, I didn't test moving python back to native after I pinned the package

File: pkgs/servers/varnish/dynamic.nix [hunk 2]
```diff
@@ -12,7 +12,7 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkgconfig docutils autoreconfHook ];
-  buildInputs = [ varnish python ];
+  buildInputs = [ varnish varnish.python ];
```
Comment (volth) [inline pkgs/servers/varnish/dynamic.nix]: so `python` is to be removed from the first line.

**Comment by jonringer** [comment]
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6Packages

I applied your suggestions

**Comment by Mic92** [comment]
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish4Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish5Packages
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.varnish6Packages

**Comment by risicle** [comment]
`nix-review` is happy on macos 10.13

**Pull request merged**